src/core/mac/mac.hpp

```diff
@@ -740,6 +746,10 @@ class Mac
 
     otMacCounters mCounters;
     uint32_t mKeyIdMode2FrameCounter;
+
+    bool mWaitingForData;
+    bool mScheduleNextTrasmission;
+    bool mFramePending;
```

Comment:
Another suggestion after thinking more about this :)
How about we use `bool mDelaySleep` instead of `bool mFramePending`? If it is set 'true`, we delay sleep.

I think frame pending name and the `Mac::GetFramePending()` and `Mac:SetFramePending()` can be confused with the ones with same name in `Frame` class.